libvast/vast/system/report.hpp

```diff
@@ -21,27 +21,50 @@
 
 namespace vast::system {
 
+/// A set of tags to attach to a metrics event.
+// This is also declared in `actors.hpp`
+using metrics_metadata = std::vector<std::pair<std::string, std::string>>;
```

Comment:
```suggestion
/// A set of tags to attach to a metrics event.
struct metrics_metadata : std::vector<std::pair<std::string, std::string>> {
  using super = std::vector<std::pair<std::string, std::string>>;
  using super::super;
};
```

Makes for way better error messages. We should use "strong" aliases for anything used in actor interfaces. Also this avoids you having to duplicate the alias in `actors.hpp`.